Tests

eyoon13 opened the pull request:
Adding Tests. HR creates error when trying to use get_employee function but ill be looking into it.

Commit:
Tests for Sales and HR wrapper

Commit:
Merge branch 'master' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

Commit:
made new tests for wrappers and correctness of data being recieved.
Errors and failures occur since some of the APIs 404. Other are issues on our end(passing json data into models).

Commit:
Cleaned up Tests

Commit:
Merge branch 'match-api' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

Commit:
commented out tests for HR since we don't have the URL to test.

Commit:
Merge branch 'master' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

paradime committed:
added travis

Commit:
Merge branch 'master' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

Commit:
updated mocked responses to match data from HR api, to make things more readable in tests. 2 errors occuring when trying to get HR information and read into array.

Commit:
Merge branches 'Tests' and 'add-travis2' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,153 @@
+        if json is None:
+            return m.get(hr.giveURL(0), json = mocked_responses.hr_get_employee )
+        else:
+            return m.get('http://vm343a.se.rit.edu:8080/employee?employee_id=0', json=json)
```
Use the `HR_URL` from the `config.py` file.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,153 @@
+from customersupport.wrappers import hr as hr_wrapper
+
+
+class Stubber:
```
Rename `Stubber` class to match the other ones, each specific to an API endpoint.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,153 @@
+            return m.get(SALES_URL, json=mocked_responses.sales_search_customer)
+        else:
+            return m.get('http://vm343c.se.rit.edu/api/customer?firstName=Joe',json=json)
+class StubberSalesOrder:
```
Please put a newline (or two, officially) above each of these classes for readability.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,153 @@
+            expected = True
+            assert actual == expected, "actual: " + str(actual.json())'''
+
+#check to see if server is alive
```
For comments like this, describing the tests, please put them as PEP8-compliant method docstrings.
[Example](https://www.python.org/dev/peps/pep-0257/#id16)

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,153 @@
+        else:
+            return m.get('http://vm343a.se.rit.edu:8080/employee?employee_id=0', json=json)
+class StubberSalesCustomer:
+    def stubMock(m, json=None):
```
This `json` parameter is unused, and is instantly overwritten by the next line.
The `if json is None` will never be `True`.

Commit:
Updated to make tests more readable, also removed Stubber Classes.

Commit:
Merge branch 'master' of https://github.com/corbanmailloux/krutzcorp-customersupport into Tests

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,97 @@
+    def test_sales_orderinfo_called(self):
+        """tests for Sales order information"""
+        try:
+            r = requests.get('http://vm343c.se.rit.edu/api/order?orderId=1&paymentInfo=False&shippingInfo=False&customerInfo=False&items=False')
```
Use the `SALES_URL` here.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,97 @@
+    def test_sales_customer_called(self):
+        """ Tests to check if search_customer was called """
+        try:
+            r = requests.get(SALES_URL+'/customer?firstName=Joe')
```
I'm not sure I understand why we are calling URLs directly in the tests. I'd think we'd be using the wrapper for that. Hence why the mocking existed initially, so we could confirm that the wrapper was calling the right URL.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -0,0 +1,97 @@
+
+class TestCase(unittest.TestCase):
+
+    def testHRMockReturnsCorrectResponse(self):
```
Match to PEP naming.
`test_hr_returns_correct_response(self)`

Commit:
updated tests to include testing our own wrappers.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -47,41 +48,31 @@ def test_sales_orderinfo(self):
-            assert r.status_code == 200
-        except ConnectionError:
-            r.raise_for_status()
+        actual = sales.get_order_info("John doe 1111 street Rochester NY 14568", True)
```
Because this method has multiple parameters, the `True` here will actually go to the `incl_payment_info` parameter. 
Please be explicit with the keyword arguments. To use the mocked version, this should be:
`actual = sales.get_order_info("John doe 1111 street Rochester NY 14568", mock=True)`

Change this and all of the others that do the same thing.

Commit:
updated tests. changed hr function names to match others.

corbanmailloux commented on customersupport/customersupport_tests.py:
```diff
@@ -48,29 +48,30 @@ def test_sales_orderinfo(self):
-        assert actual == expected
+        actual = sales.get_order_info("John doe 1111 street Rochester NY 14568", mock=True)
+        final = [actual[0].id,actual[0].order_date]
+        expected = expected = [1000, '2017-03-01T20:51:26.905Z']
```
`expected = expected = [...]`?

corbanmailloux reviewed (commented):
I'll run these tonight and double check, but it looks good.

Commit:
corrected tests(HR was trying to compare to json instead of the employee object).

corbanmailloux commented:
Also, please document the correct way to run the tests. Do I run the tests file directly or use `python -m unittest discover`, for example?

Commit:
1. Updated README to explain how to run Tests
2. Renamed testing file to tests.py
3. Moved File up one directory to allow access to needed modules.

corbanmailloux commented:
This looks really good. On my machine, some of these tests are currently erroring. I think it may be due to API changes, but I can't look into it at the moment.

Commit:
1. updated readme and made on last pull from master before merge.
